HLTrigger/JetMET/src/HLTDiJetEtaTopologyFilter.cc

```diff
@@ -0,0 +1,169 @@
+/** \class HLTDiJetEtaTopologyFilter
+ *
+ *
+ *  \author Tomasz Fruboes
+ *     based on HLTDiJetAveFilter
+ */
+
+#include "HLTrigger/JetMET/interface/HLTDiJetEtaTopologyFilter.h"
+
+#include "DataFormats/Common/interface/Ref.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Math/interface/deltaPhi.h"
+
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+#include "HLTrigger/HLTcore/interface/defaultModuleLabel.h"
+
+#include<typeinfo>
+
+//
+// constructors and destructor
+//
+template<typename T>
+HLTDiJetEtaTopologyFilter<T>::HLTDiJetEtaTopologyFilter(const edm::ParameterSet& iConfig) : HLTFilter(iConfig),
+    inputJetTag_ (iConfig.template getParameter< edm::InputTag > ("inputJetTag")),
+    minPtAve_    (iConfig.template getParameter<double> ("minPtAve")),
+    atLeastOneJetAbovePT_(iConfig.template getParameter<double> ("atLeastOneJetAbovePT")),
+    minPtTag_(iConfig.template getParameter<double> ("minPtTag")),
+    minPtProbe_(iConfig.template getParameter<double> ("minPtProbe")),
+    minDphi_     (iConfig.template getParameter<double> ("minDphi")),
+    tagEtaMin_     (iConfig.template getParameter<double> ("minTagEta")),
+    tagEtaMax_     (iConfig.template getParameter<double> ("maxTagEta")),
+    probeEtaMin_     (iConfig.template getParameter<double> ("minProbeEta")),
+    probeEtaMax_     (iConfig.template getParameter<double> ("maxProbeEta")),
+    applyAbsToTag_ (iConfig.template getParameter<bool> ("applyAbsToTag")),
+    applyAbsToProbe_ (iConfig.template getParameter<bool> ("applyAbsToProbe")),
+    oppositeEta_ (iConfig.template getParameter<bool> ("oppositeEta")),
+
+    triggerType_ (iConfig.template getParameter<int> ("triggerType"))
+{
+    m_theJetToken = consumes<std::vector<T>>(inputJetTag_);
```

Comment:
... by moving this line up into the initialiser list via
```
m_theJetToken_(consumes<std::vector<T>>(inputJetTag_)),
```
after the line where inputJetTag_ is initialised.